zkevm-circuits/src/witness/tx.rs

```diff
@@ -725,14 +954,53 @@ impl Transaction {
                     bytes_rlc,
                     gas_cost_acc,
                 },
+                // The stack operations will be later sorted and assigned
+                rlp_decoding_table: RlpStackOp::default(),
             });
+
             witness_table_idx += 1;
 
-            if cur.tag == EndList && cur.depth == 0 {
+            if cur.tag == EndObject && cur.depth == 0 {
                 break;
             }
             cur = next;
         }
+
+        assert_eq!(
+            stack_ops.len(),
+            witness.len(),
+            "Number of stack_ops must be equal to witness length"
+        );
+
+        // Sort the RlpStackOps and assign to the RlpDecodingTable part of witness
+        stack_ops.sort_by(|a, b| {
+            if (
+                a.tx_id,
+                a.format as u64,
+                a.depth,
+                a.byte_idx,
```

Comment:
why does `byte_idx` have higher precedence than `al_idx` and `sk_idx`?

Reply:
`byte_idx` is kind of like `rw_counter` in state circuit.

Reply:
Corrected.